experimental/Schemes/RationalMap.jl

```diff
@@ -535,70 +544,108 @@ function pullback(phi::RationalMap, C::AbsAlgebraicCycle)
     sort!(all_U, lt=(x,y)->complexity(x)<complexity(y))
 
     # First try to get hold of the component via cheap realizations 
-    success = false
     for U in all_U
       psi = cheap_realization(phi, U, V)
       U_sub = domain(psi)
       J = pullback(psi)(saturated_ideal(I(V)))
       if !isone(J)
         JJ = IdealSheaf(X, domain(psi), gens(J))
-        comps[JJ] = C[I]
-        success = true
+        return JJ
         break
       end
     end
+  end
+  return nothing
+end
 
-    success && break
-
+function _try_randomized_pullback(phi::RationalMap, I::IdealSheaf)
+  X = domain(phi)
+  Y = codomain(phi)
+  scheme(I) === Y || error("ideal sheaf not defined on the correct scheme")
+  # Find a patch in Y on which this component is visible
+  all_V = [V for V in affine_charts(Y) if !isone(I(V))]
+
+  @show all_V
+  min_var = minimum([ngens(OO(V)) for V in all_V])
+  @show min_var
+  all_V = [V for V in all_V if ngens(OO(V)) == min_var]
+  @show all_V
+  deg_bound = minimum([maximum([total_degree(lifted_numerator(g)) for g in gens(I(V))]) for V in all_V])
+  @show deg_bound
+  all_V = [V for V in all_V if minimum([total_degree(lifted_numerator(g)) for g in gens(I(V))]) == deg_bound]
+  @show all_V
+  V = first(all_V)
+  @show findfirst(x->x===V, affine_charts(Y))
+
+  all_U = copy(affine_charts(X))
+  function complexity(U::AbsSpec)
+    a = realization_preview(phi, U, V)
+    return maximum(vcat([total_degree(numerator(f)) for f in a], [total_degree(denominator(f)) for f in a]))
+  end
+  sort!(all_U, lt=(x,y)->complexity(x)<complexity(y))
```

Comment:
This should also work (untested)
```suggestion
  sort!(all_U, by=complexity)
```